libfreerdp/core/connection.c

```diff
@@ -1525,7 +1525,11 @@ BOOL rdp_server_accept_nego(rdpRdp* rdp, wStream* s)
 	SelectedProtocol = nego_get_selected_protocol(nego);
 	status = FALSE;
 
-	if (SelectedProtocol & PROTOCOL_RDSTLS)
+	if (freerdp_settings_get_bool(rdp->settings, FreeRDP_VmConnectMode) &&
+		SelectedProtocol != PROTOCOL_RDP)
```

Comment:
so even `TLS` security? that is more or less the same as `RDP` except it is using standard `anonymous TLS` instead of `MS` grown custom `RSA` stuff

Reply:
Yes, I just doubled check. For anything but RDP, vmms.exe intercepts it.
Since vmms.exe is basically doing TLS interception, it somewhat makes sense for it not to re-create a TLS connection just to talk to a local VM.